scipy/sparse/_coo.py

```diff
@@ -666,54 +681,95 @@ def _matmul_dispatch(self, other):
             if other.ndim == 1:
                 msg = f"{err_prefix} (n,k={N}),(k={other.shape[0]},)->(n,)"
                 raise ValueError(msg)
-            msg = "n-D matrix-matrix multiplication not implemented for ndim>2"
-            raise NotImplementedError(msg)
-
+            if other.shape[-2] == N:
+                # check for batch dimensions compatibility
+                batch_shape_A = self.shape[:-2]
+                batch_shape_B = other.shape[:-2]
+                if batch_shape_A != batch_shape_B:
+                    try:
+                        # This will raise an error if the shapes are not broadcastable
+                        np.broadcast_shapes(batch_shape_A, batch_shape_B)
+                    except ValueError:
+                        raise ValueError("Batch dimensions are not broadcastable")
+            
+                return self._matmul_multivector(other)
+            else:
+                raise ValueError('could not interpret dimensions')
+        
+            
         if isscalarlike(other):
             # scalar value
             return self._mul_scalar(other)
 
         if issparse(other):
-            raise NotImplementedError("sparse-sparse matmul not implemented for ndim>2")
-
-        # If it's a list or whatever, treat it like an array
-        other_a = np.asanyarray(other)
+            self_is_1d = False
+            other_is_1d = False
```

Comment:
Let's use 
```python
            self_is_1d = self.ndim == 1
            other_is_1d = self.other == 1
```
and for `self` we can use `self._shape_as_2d` to add the 1 in front. `other` puts the 1 on the other side so looks good here.

Reply:
Is this alright at 09c4bf2a200fa75a367810257c2c9d461730e092?